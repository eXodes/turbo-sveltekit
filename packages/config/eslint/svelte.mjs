import { rules } from "./rules.mjs";
import prettier from "eslint-config-prettier";
import turbo from "eslint-config-turbo/flat";
import onlyWarn from "eslint-plugin-only-warn";
import svelte from "eslint-plugin-svelte";
import unicorn from "eslint-plugin-unicorn";
import globals from "globals";
import svelteParser from "svelte-eslint-parser";
import ts from "typescript-eslint";

export default ts.config(
  unicorn.configs["flat/recommended"],
  ...ts.configs.recommended,
  ...ts.configs.stylistic,
  ...svelte.configs["flat/recommended"],
  prettier,
  ...svelte.configs["flat/prettier"],
  ...turbo,
  {
    plugins: {
      "only-warn": onlyWarn,
    },

    languageOptions: {
      globals: { ...globals.browser, ...globals.node },

      ecmaVersion: "latest",
      sourceType: "module",

      parser: ts.parser,
      parserOptions: {
        tsconfigRootDir: import.meta.dirname,
        extraFileExtensions: [".svelte"],
      },
    },

    rules: {
      ...rules,
    },
  },
  {
    files: ["**/*.svelte"],

    languageOptions: {
      parser: svelteParser,
      parserOptions: {
        parser: ts.parser,
      },
    },
  }
);
