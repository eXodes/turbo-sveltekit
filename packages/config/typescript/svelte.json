{
  "display": "Svelte",
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "lib": ["esnext", "dom", "dom.iterable"],
    "module": "esnext",
    "moduleDetection": "force",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "verbatimModuleSyntax": true,
    "useDefineForClassFields": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext"
  }
}
