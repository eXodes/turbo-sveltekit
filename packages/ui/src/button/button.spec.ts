import Button from "./button.svelte";
import { render, screen } from "@testing-library/svelte";
import { userEvent } from "@testing-library/user-event";
import { createRawSnippet } from "svelte";
import { describe, expect, test, vi } from "vitest";

describe("Button", () => {
  const children = createRawSnippet(() => ({
    render: () => `<span>Button</span>`,
  }));

  test("should render default button", async () => {
    const { container } = render(Button, {
      children,
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should render primary button", async () => {
    const { container } = render(Button, {
      children,
      color: "primary",
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should render small button", async () => {
    const { container } = render(Button, {
      children,
      size: "small",
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should render medium button", async () => {
    const { container } = render(Button, {
      children,
      size: "medium",
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should render large button", async () => {
    const { container } = render(Button, {
      children,
      size: "large",
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should be able to click button", async () => {
    const onclick = vi.fn();

    render(Button, {
      children,
      onclick,
    });

    const ButtonEl = screen.getByRole("button");

    await userEvent.click(ButtonEl);

    expect(onclick).toHaveBeenCalledTimes(1);
  });
});
