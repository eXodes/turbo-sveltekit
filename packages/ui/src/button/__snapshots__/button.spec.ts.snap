// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render default button 1`] = `
HTMLCollection [
  <button
    class="inline-flex min-h-10 cursor-pointer items-center justify-center rounded-lg font-semibold shadow-md transition active:scale-98 bg-gray-200 text-gray-800 hover:bg-gray-100 shadow-gray-300/75 px-4 py-2 text-sm"
    data-button-root="true"
    tabindex="0"
  >
    <!---->
    <!---->
    <span>
      Button
    </span>
    
    
    
  </button>,
]
`;

exports[`Button > should render large button 1`] = `
HTMLCollection [
  <button
    class="inline-flex min-h-10 cursor-pointer items-center justify-center rounded-lg font-semibold shadow-md transition active:scale-98 bg-gray-200 text-gray-800 hover:bg-gray-100 shadow-gray-300/75 px-5 py-3 text-base"
    data-button-root="true"
    size="large"
    tabindex="0"
  >
    <!---->
    <!---->
    <span>
      Button
    </span>
    
    
    
  </button>,
]
`;

exports[`Button > should render medium button 1`] = `
HTMLCollection [
  <button
    class="inline-flex min-h-10 cursor-pointer items-center justify-center rounded-lg font-semibold shadow-md transition active:scale-98 bg-gray-200 text-gray-800 hover:bg-gray-100 shadow-gray-300/75 px-4 py-2 text-sm"
    data-button-root="true"
    size="medium"
    tabindex="0"
  >
    <!---->
    <!---->
    <span>
      Button
    </span>
    
    
    
  </button>,
]
`;

exports[`Button > should render primary button 1`] = `
HTMLCollection [
  <button
    class="inline-flex min-h-10 cursor-pointer items-center justify-center rounded-lg font-semibold shadow-md transition active:scale-98 bg-orange-600 text-orange-50 hover:bg-orange-500 shadow-orange-500/75 px-4 py-2 text-sm"
    color="primary"
    data-button-root="true"
    tabindex="0"
  >
    <!---->
    <!---->
    <span>
      Button
    </span>
    
    
    
  </button>,
]
`;

exports[`Button > should render small button 1`] = `
HTMLCollection [
  <button
    class="inline-flex min-h-10 cursor-pointer items-center justify-center rounded-lg font-semibold shadow-md transition active:scale-98 bg-gray-200 text-gray-800 hover:bg-gray-100 shadow-gray-300/75 px-3 py-1.5 text-xs"
    data-button-root="true"
    size="small"
    tabindex="0"
  >
    <!---->
    <!---->
    <span>
      Button
    </span>
    
    
    
  </button>,
]
`;
